R/teal_slice.R

```diff
@@ -334,6 +329,60 @@ is.teal_slices <- function(x) { # nolint
 }
 
 
+# utils -----------------------------------------------------------------------------------------------------------
+
+trim_character <- function(x, nchar) {
+  x_t <- substr(x, 1, nchar)
+  substr(x_t, nchar - 2, nchar) <- "..."
+  x_t
+}
+
+justify_json_split <- function(json, format_fun) {
+  json_s <- strsplit(json, split = ":", fixed = TRUE)
+  name_width <- max(unlist(gregexpr(":", json))) - 1
+  vapply(json_s, function(x) paste0(format_fun(x[1], name_width), na.omit(x[2])), character(1))
+}
+
+# JSON utils for teal_slice ---------------------------------------------------------------------------------------
+
+to_json <- function(x) {
+  no_unbox <- function(x) {
+    nms <- c("selected", "choices")
+    if (is.list(x)) {
+      for (nm in nms) {
+        if (!is.null(x[[nm]])) x[[nm]] <- I(x[[nm]])
+      }
+      lapply(x, no_unbox)
+    } else {
+      x
+    }
+  }
+
+  jsonlite::toJSON(no_unbox(x), pretty = TRUE, auto_unbox = TRUE, digits = 16, null = "null")
+}
+
+slice_format_name <- function(n, name_width) {
+  if (nchar(n) == 1) {
+    return(n)
+  } else {
+    paste(format(n, width = name_width), ":")
+  }
+}
+
+# JSON utils for teal_slices --------------------------------------------------------------------------------------
+
+slices_format_name <- function(n, name_width) {
+  if (nchar(gsub("\\s", "", n)) <= 2) {
+    return(n)
+  } else if (grepl("slices|attributes", n)) {
+    paste0(n, ":")
+  } else {
+    paste(format(n, width = name_width), ":")
+  }
+}
```

Comment:
let's make them inline function in the place where they are called (`format`). Otherwise I'm good with everything else. Please update roxygen, pkgdown.yaml (by new exported functions) and tests

Reply:
f7f273982368b5210ba334c359fab981b9ce0bf2